### PR TITLE
Introduce RoverCommander.getOrElseFetchBestArticle

### PR DESCRIPTION
**Pull request by leogrim**

### CODE REVIEW

File: kifi-backend/modules/rover/app/com/keepit/rover/store/RoverArticleStore.scala [hunk 1]
```diff
@@ -20,12 +19,12 @@ class RoverArticleStore @Inject() (underlying: RoverUnderlyingArticleStore, priv
...
 
-  def get[A <: Article](key: ArticleKey[A])(implicit classTag: ClassTag[A]): Future[Option[A]] = {
+  def get[A <: Article](key: ArticleKey[A]): Future[Option[A]] = {
     consolidate(key)(storeKey => SafeFuture {
       underlying.syncGet(storeKey)
     }).imap { articleOpt =>
       articleOpt.map {
-        case expectedArticle: A => expectedArticle
+        case expectedArticle if expectedArticle.kind == key.kind => expectedArticle.asInstanceOf[A]
         case unexpectedArticle => throw new InconsistentArticleTypeException(key, unexpectedArticle)
```
Comment (leogrim) [inline kifi-backend/modules/rover/app/com/keepit/rover/store/RoverArticleStore.scala]: @andrewconner @stkem I give up, I'm ready to go to hell for this.

Comment (stkem) [inline kifi-backend/modules/rover/app/com/keepit/rover/store/RoverArticleStore.scala]: huh?

Comment (andrewconner) [inline kifi-backend/modules/rover/app/com/keepit/rover/store/RoverArticleStore.scala]: Poor guy.

Comment (andrewconner) [inline kifi-backend/modules/rover/app/com/keepit/rover/store/RoverArticleStore.scala]: Actually this is perfect for Typesafe support. Mind minimizing a reproduction, and I'll ask them?

**Comment by leogrim** [comment]
@camhashemi

**Pull request merged**